lib/connectionConfig/connectionConfig.ts

```diff
@@ -112,25 +111,38 @@ export namespace ConnectionConfig {
    */
   export function validate(config: ConnectionConfig, options?: ConnectionConfigOptions): void {
     if (!options) options = {};
-    if (!config || (config && typeof config !== "object")) {
-      throw new Error("'config' is a required parameter and must be of type: 'object'.");
+
+    if (!config) {
+      throw new TypeError("Missing configuration");
+    }
+
+    if (!config.endpoint) {
+      throw new TypeError("Missing 'endpoint' in configuration");
     }
-    if (!config.endpoint || (config.endpoint && typeof config.endpoint !== "string")) {
-      throw new Error("'endpoint' is a required property of ConnectionConfig.");
+    config.endpoint = String(config.endpoint);
+
+    if (config.entityPath) {
+      config.entityPath = String(config.entityPath);
     }
-    if (config.entityPath && typeof config.entityPath !== "string") {
-      throw new Error("'entityPath' must be of type 'string'.");
+
+    if (config.host) {
+      config.host = String(config.host);
```

Comment:
why is it that we throw error for missing endpoint, but not for missing `host`?

Reply:
I'll add the missing host check.